ads/google/doubleclick.js

```diff
@@ -138,77 +93,22 @@ function doubleClickWithGpt(global, data, gladeExperiment) {
-  //    form WxH;
-  // 2) that each secondary dimension is strictly less than its primary
-  //    dimension counterpart;
-  // 3) and, if data-mutli-size-validation is not set to false, that each
-  //    secondary dimension is at least 2/3rds of its primary dimension
-  //    counterpart.
+  // Handle multi-size data parsing, validation, and inclusion into dimensions.
   const multiSizeDataStr = data.multiSize || null;
   if (multiSizeDataStr) {
-    const sizesStr = multiSizeDataStr.split(',');
-    sizesStr.forEach(sizeStr => {
-
-      const size = sizeStr.split('x');
-
-      // Make sure that each size is specified in the form val1xval2.
-      if (size.length != 2) {
-        user().error('AMP-AD', `Invalid multi-size data format '${sizeStr}'.`);
-        return;
-      }
-
-      const widthStr = size[0];
-      const heightStr = size[1];
-
-      // Make sure that both dimensions given are numbers.
-      if (!validateDimensions(widthStr, heightStr,
-            w => isNaN(Number(w)),
-            h => isNaN(Number(h)),
-            ({badDim, badVal}) =>
-            `Invalid ${badDim} of ${badVal} given for secondary size.`)) {
-        return;
-      }
-
-      const width = Number(widthStr);
-      const height = Number(heightStr);
-      const primarySize = dimensions[0];
-      const primaryWidth = primarySize[0];
-      const primaryHeight = primarySize[1];
-
-      // Check that secondary size is not larger than primary size.
-      if (!validateDimensions(width, height,
-            w => w > primaryWidth,
-            h => h > primaryHeight,
-            ({badDim, badVal}) => `Secondary ${badDim} ${badVal} ` +
-              `can't be larger than the primary ${badDim}.`)) {
-        return;
-      }
-
-      // Check that if multi-size-validation is on, that the secondary sizes
-      // are at least minRatio of the primary size.
-      const validate = data.multiSizeValidation || 'true';
-      if (validate != 'false' && validate != false) {
-
-        // The minimum ratio of each secondary dimension to its corresponding
-        // primary dimension.
-        const minRatio = 2 / 3;
-        const minWidth = minRatio * primaryWidth;
-        const minHeight = minRatio * primaryHeight;
-        if (!validateDimensions(width, height,
-              w => w < minWidth,
-              h => h < minHeight,
-              ({badDim, badVal}) => `Secondary ${badDim} ${badVal} is ` +
-              `smaller than 2/3rds of the primary ${badDim}.`)) {
-          return;
-        }
-      }
-
-      // Passed all checks! Push additional size to dimensions.
-      dimensions.push([width, height]);
-    });
+    const primarySize = dimensions[0];
+    const primaryWidth = primarySize[0];
+    const primaryHeight = primarySize[1];
+    // By default, this flag is true.
+    const multiSizeValidation = data.multiSizeValidation ?
+        data.multiSizeValidation == 'true' : true;
+
+    getMultiSizeDimensions(
```

Comment:
If the return is not actually being used anywhere, unless you have a known use case for it that is being implemented, I think it would be clearer to simply not return anything. In the future, if someone ends up needing it, they can make the relevant change